lib/waterline/query/finders/basic.js

```diff
@@ -45,6 +46,10 @@ module.exports = {
     // Normalize criteria
     criteria = normalize.criteria(criteria);
 
+    this.beforeCallbacks.call(self,criteria.where,function(err){
+      if (err) {return cb(err);}
+    });
```

Comment:
Looking at how `beforeDestroy` and `beforeUpdate` work, the code flow is much nicer. They appropriately handle all async callbacks and such. It feels like the right thing to do would be to put everything after this inside the callback. That way folks could do async operations if they wanted to. 

Would up the line count change significantly, but would just be indentation.